src/interactive-window/interactiveWindow.ts

```diff
@@ -95,35 +87,30 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
     public get submitters(): Uri[] {
         return this._submitters;
     }
-    public get notebookUri(): Uri | undefined {
-        return this._notebookDocument?.uri;
+    public get notebookUri(): Uri {
```

Comment:
Lots of nullable removed, it was actually impossible to have nullables, but we had it because of the way we structured the code. Now the editor is created outside in the factory class and injected into this.

nulls can be a pain.